ContextMenuTests.SharePopoverDoesNotClearSelection times out in some versions of macOS

whsieh opened the pull request:
#### ab2e98f51ffb9d9315b44fd88598ac2c68fac9a4
<pre>
ContextMenuTests.SharePopoverDoesNotClearSelection times out in some versions of macOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=251690">https://bugs.webkit.org/show_bug.cgi?id=251690</a>
rdar://104987312

Reviewed by Aditya Keerthi.

After some refactoring in system share sheet, `SHKSharingServicePicker` no longer shows a popover if
it can&apos;t find a presenting view, by asking the menu item&apos;s `NSMenu` for its `-_presentingView`.
However, in the case where the menu item is programmatically invoked, the presenting view seems to
be `nil`, which breaks this API test, since it needs to verify that presenting the share sheet
popover does not erroneously clear out the active text selection.

To keep this test passing after these recent system changes, we swizzle `-[NSMenu _presentingView]`
to force the share sheet popover to appear next to the web view.

* Tools/TestWebKitAPI/Tests/TestWebKitAPI/mac/AppKitSPI.h:
* Tools/TestWebKitAPI/Tests/mac/ContextMenuTests.mm:
(TestWebKitAPI::TEST):

Canonical link: <a href="https://commits.webkit.org/259834@main">https://commits.webkit.org/259834@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eac47bc17427f730b3226330cdc5ed03b3fe2073

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/106071 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/15126 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/38902 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/115252 "Built successfully") | [❌ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/175345 "Found 1 new test failure: http/tests/fetch/redirectmode-and-preload.html (failure)") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/109973 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/16552 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/6295 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/98279 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/114969 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/111824 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/12603 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95576 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/40140 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/94475 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/27219 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/81818 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/8376 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28571 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/8869 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/5118 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/14489 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/48115 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6803 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/10413 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder reviewed (approved):
Seems like a reasonable workaround – though we should follow up with ShareKit to confirm they didn't break a real use case.

whsieh committed:
ContextMenuTests.SharePopoverDoesNotClearSelection times out in some versions of macOS
https://bugs.webkit.org/show_bug.cgi?id=251690
rdar://104987312

Reviewed by Aditya Keerthi.

After some refactoring in system share sheet, `SHKSharingServicePicker` no longer shows a popover if
it can't find a presenting view, by asking the menu item's `NSMenu` for its `-_presentingView`.
However, in the case where the menu item is programmatically invoked, the presenting view seems to
be `nil`, which breaks this API test, since it needs to verify that presenting the share sheet
popover does not erroneously clear out the active text selection.

To keep this test passing after these recent system changes, we swizzle `-[NSMenu _presentingView]`
to force the share sheet popover to appear next to the web view.

* Tools/TestWebKitAPI/Tests/TestWebKitAPI/mac/AppKitSPI.h:
* Tools/TestWebKitAPI/Tests/mac/ContextMenuTests.mm:
(TestWebKitAPI::TEST):

Canonical link: https://commits.webkit.org/259834@main
